lib/arjdbc/postgresql/base/array_encoder.rb

```diff
@@ -0,0 +1,21 @@
+# This implements a basic encoder to work around ActiveRecord's dependence on the pg gem
+module PG
```

Comment:
I think forward progress here and in your other comments is probably a bit more important than stalling to do this differently.  If @kares has any insight we will just follow up on it.